src/geometry/manifold/manifold.lean

```diff
@@ -0,0 +1,531 @@
+  simp only [set.univ_inter, set.preimage_univ]
+end
+
+lemma open_target (he : e ∈ c.atlas) : is_open e.target :=
+begin
+  have E : e.target ∩ e.inv_fun ⁻¹' e.source = e.target :=
+  subset.antisymm (inter_subset_left _ _) (λx hx, ⟨hx,
+    local_equiv.target_subset_preimage_source _ hx⟩),
+  simpa [local_equiv.trans_source, E] using c.open_source e e he he
+end
+
+def local_homeomorph (e : local_equiv M H) (he : e ∈ c.atlas) :
+  @local_homeomorph M H c.to_topological_space _ :=
+{ open_source := by convert c.open_source' he,
+  open_target := by convert c.open_target he,
+  continuous_to_fun := begin
+    letI : topological_space M := c.to_topological_space,
+    rw continuous_on_open_iff (c.open_source' he),
+    assume s s_open,
+    rw inter_comm,
+    apply topological_space.generate_open.basic,
+    simp only [exists_prop, mem_Union, mem_singleton_iff],
+    exact ⟨e, he, ⟨s, s_open, rfl⟩⟩
+  end,
+  continuous_inv_fun := begin
+    letI : topological_space M := c.to_topological_space,
+    apply continuous_on_open_of_generate_from (c.open_target he),
+    assume t ht,
+    simp only [exists_prop, mem_Union, mem_singleton_iff] at ht,
+    rcases ht with ⟨e', e'_atlas, s, s_open, ts⟩,
+    rw ts,
+    let f := e.symm.trans e',
+    have : is_open (f.to_fun ⁻¹' s ∩ f.source),
+      by simpa [inter_comm] using (continuous_on_open_iff (c.open_source e e' he e'_atlas)).1
+        (c.continuous_to_fun e e' he e'_atlas) s s_open,
+    have A : e'.to_fun ∘ e.inv_fun ⁻¹' s ∩ (e.target ∩ e.inv_fun ⁻¹' e'.source) =
+             e.target ∩ (e'.to_fun ∘ e.inv_fun ⁻¹' s ∩ e.inv_fun ⁻¹' e'.source),
+      by { rw [← inter_assoc, ← inter_assoc], congr' 1, exact inter_comm _ _ },
+    simpa [local_equiv.trans_source, preimage_inter, preimage_comp.symm, A] using this
+  end,
+  ..e }
+
+def to_manifold : @manifold H _ M c.to_topological_space :=
+{ atlas := ⋃ (e : local_equiv M H) (he : e ∈ c.atlas), {c.local_homeomorph e he},
+  chart_at := λx, c.local_homeomorph (c.chart_at x) (c.chart_mem_atlas x),
+  mem_chart_source := λx, c.mem_chart_source x,
+  chart_mem_atlas := λx, begin
+    simp only [mem_Union, mem_singleton_iff],
+    exact ⟨c.chart_at x, c.chart_mem_atlas x, rfl⟩,
+  end }
+
+end manifold_core
+
+section has_groupoid
+
+/-- A manifold has an atlas in a groupoid G if the change of coordinates belong to the groupoid -/
+class has_groupoid (H : Type*) [topological_space H] (M : Type*) [topological_space M]
+  [manifold H M] (G : structure_groupoid H) : Prop :=
+(compatible : ∀e e' : local_homeomorph M H, e ∈ atlas M H → e' ∈ atlas M H → (e.symm →ₕ e') ∈ G)
+
+lemma has_groupoid_of_le {H : Type*} [topological_space H] {M : Type*} [topological_space M]
+  [manifold H M] {G₁ G₂ : structure_groupoid H} (h : has_groupoid H M G₁) (hle : G₁ ≤ G₂) :
+  has_groupoid H M G₂ :=
+⟨ λ e e' he he', hle ((h.compatible : _) e e' he he') ⟩
+
+/-- The trivial manifold structure on the model space is compatible with any groupoid -/
```

Comment:
I guess dually, every manifold is compatible with the continuous groupoid `\top`.

Reply:
Yes, good remark. I have added this.